src/main/java/picard/fingerprint/CheckFingerprint.java

```diff
@@ -129,37 +134,65 @@ protected int doWork() {
             outputSummaryMetricsFile = new File(OUTPUT + FINGERPRINT_SUMMARY_FILE_SUFFIX);
         }
 
-        IOUtil.assertFileIsReadable(INPUT);
         IOUtil.assertFileIsReadable(HAPLOTYPE_MAP);
         IOUtil.assertFileIsReadable(GENOTYPES);
         IOUtil.assertFileIsWritable(outputDetailMetricsFile);
         IOUtil.assertFileIsWritable(outputSummaryMetricsFile);
 
         final FingerprintChecker checker = new FingerprintChecker(HAPLOTYPE_MAP);
+        List<FingerprintResults> results;
 
-        SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), SAMSequenceDictionaryExtractor.extractDictionary(GENOTYPES), true);
-        SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), checker.getHeader().getSequenceDictionary(), true);
+        String observedSampleAlias = null;
+        final boolean isBamOrSamFile = isBamOrSamFile(INPUT);
+        if (isBamOrSamFile) {
+            SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), SAMSequenceDictionaryExtractor.extractDictionary(GENOTYPES), true);
+            SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), checker.getHeader().getSequenceDictionary(), true);
 
-        // If sample alias isn't supplied, assume it's the one from the INPUT file's RGs
-        if (SAMPLE_ALIAS == null) {
+            // Verify that there's only one sample in the SAM/BAM.
```

Comment:
No.  I still want to validate that there's only one sample in the bam, so that's done in the for loop starting at line 154.  If there's more than one sample in the BAM, then an exception will be thrown, otherwise, observedSampleAlias will be defined.

The if statement at 165 checks if the user set the EXPECTED_SAMPLE_ALIAS to null and if so sets it to the observedSampleAlias I set as a result of the for loop above.